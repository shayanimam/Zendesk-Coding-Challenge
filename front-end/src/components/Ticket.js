import React, { useState } from "react";
import Dialog from "@mui/material/Dialog";

const Ticket = ({ id, subject, description, status, created_at }) => {
  const date = new Date(created_at).toLocaleString("en-US");
  const [expand, setExpand] = useState(false);

  const toggleDialog = () => {
    setExpand((prev) => !prev);
  };

  return (
    <div
      className="ticket-wrapper"
      onClick={toggleDialog}
      data-testid="ticket-wrapper"
    >
      <Dialog
        open={expand}
        onClose={toggleDialog}
        onBackdropClick={toggleDialog}
        fullWidth
        className="dialog"
      >
        <div className="ticket-header">
          <h3>ID: {id}</h3>
          <h3>{subject}</h3>
          <h3>[{status}]</h3>
        </div>
        <h4 className="ticket-description">{description}</h4>
        <h4 className="ticket-date">Date created: {date}</h4>
      </Dialog>
      <div className="list-view">
        <h4>#{id}</h4>
        <h4>{subject}</h4>
      </div>
    </div>
  );
};
export { Ticket };
