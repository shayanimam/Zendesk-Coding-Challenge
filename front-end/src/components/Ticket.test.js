import { Ticket } from "./Ticket";
import { fireEvent, render } from "@testing-library/react";
import "@testing-library/jest-dom";

const defaultProps = {
  id: 1,
  subject: "abcd",
  description: "efg",
  status: "open",
  created_at: "today",
};

describe("Ticket", () => {
  it("renders", () => {
    const { container } = render(<Ticket {...defaultProps} />);

    expect(container.firstChild).toMatchSnapshot();
  });
  it("opens a dialog when clicked", () => {
    const { getByText } = render(<Ticket {...defaultProps} />);

    const ticket = getByText("abcd");
    fireEvent.click(ticket);
    const description = getByText("efg");

    expect(description).toBeVisible();
  });
});
