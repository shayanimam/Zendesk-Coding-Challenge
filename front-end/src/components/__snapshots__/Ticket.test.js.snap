// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ticket renders 1`] = `
<div
  class="ticket-wrapper"
  data-testid="ticket-wrapper"
>
  <div
    class="list-view"
  >
    <h4>
      #
      1
    </h4>
    <h4>
      abcd
    </h4>
  </div>
</div>
`;
